content/blog/monthly-july-19.md

```diff
@@ -0,0 +1,133 @@
+
+Jepsen tested TiDB and wrote a detailed [report](https://pingcap.com/blog/tidb-passes-jepsen-test-for-snapshot-isolation-and-single-key-linearizability/). This focussed on transaction guarantees and covered TiKV as the underlying system.
+
+We [released](https://pingcap.com/blog/database-cluster-deployment-and-management-made-easy-with-kubernetes/) 1.0 of TiDB Operator, a tool for deploying TiDB in the cloud.
+
+
+## Reading material
+
+Some blog posts and articles related to TiKV:
+
+* TiKV launched a [training course](https://tikv.org/blog/talent-training/) covering Rust and distributed systems.
+* Nick Cameron wrote about migrating the Rust TiKV client from [futures 0.1 to 0.3](https://www.ncameron.org/blog/migrating-a-crate-from-futures-0-1-to-0-3/).
+* Ana Hobden wrote on the PingCAP blog about [benchmarking distributed databases](https://www.pingcap.com/blog/why-benchmarking-distributed-databases-is-so-hard/).
+
+
+## Current projects
+
+There's lots going on in the TiKV ecosystem, here we'll try to highlight some of the interesting projects.
+
+### Performance
+
+There's ongoing work to measure and improve TiKV's performance in all areas. We hope to report more on specifics soon.
+
+
+### Titan
+
+[Titan](https://github.com/pingcap/titan) is a storage engine which improves on RocksDB. It is part of TiKV already and you can try it out by enabling it in the config file:
+
+```
+[rocksdb.titan]
+enabled = true
+```
+
+Note that you can't go back to RocksDB after enabling Titan. We're working to improve its performance and stability. You can read more in this [blog post](https://pingcap.com/blog/titan-storage-engine-design-and-implementation/).
+
+
+### Docs
+
+We are currently migrating [our documentation](https://tikv.org/docs/3.0/concepts/overview/) from the TiKV's wiki to the official website.
+
+
+### Replica read
+
+Replica read allows a database to read from a Raft follower, not just the leader. This greatly improves read performance when the client is close to a follower, but far from the leader (e.g., in different data centres). This feature is implemented in TiKV and we are working on adding support to TiDB.
+
+
+### Protobuf implementation
+
+We're experimenting with different implementations of Protocol Buffers (which underpin all our RPC communication). We've added support for [Prost](https://github.com/danburkert/prost) to our supporting libraries, and are working on adding support to TiKV.
+
+
+### Testing and stability
+
+We're working on making testing more reliable, adding long-running tests to our CI, and monitoring performance.
+
+
+### Adding RPN functions
+
+TiKV supports executing SQL functions on the database using its coprocessor. We've been working to support RPN (reverse polish notation) functions, which are an optimisation technique for coprocessor. In the past month we've added support for `coalesce`, `in`, `case_when`, `if`, `abs`, `multiply`, `divide`, and `date_format`.
```

Comment:
Maybe we could point them to https://pingcap.com/blog/adding-built-in-functions-to-tikv/ if they want to contribute?